### PR TITLE
Fix localtime implementation

### PR DESCRIPTION
**Pull request by appcypher**
### What does this PR do?
- Fixes the issue with localtime and localtime_r
- Adds implementation for asctime

### How should this be manually tested?
- Compile a simple C file (that prints `argv` and `argc` to the console) using Emscripten.
  ```bash
  $ emcc simple.c -o simple.js
  ```
- Run wasmer on the generated wasm file
  ```bash
  $ wasmer run simple.wasm
  ```

### Background context
- N/A

### Issues
- N/A

### CODE REVIEW

File: src/apis/emscripten/mod.rs [hunk 1]
```diff
@@ -44,11 +46,10 @@ fn dynamictop_ptr(static_bump: u32) -> u32 {
...
 
-// fn static_alloc(size: usize, static_top: &mut size) -> usize {
-//     let ret = *static_top;
-//     *static_top = (*static_top + size + 15) & (-16 as usize);
-//     ret
-// }
+
+pub fn statictop(static_bump: u32) -> u32 {
+    GLOBAL_BASE + 5520
+}
```
Comment (syrusakbary) [inline src/apis/emscripten/mod.rs]: I'm not sure if this is a good idea. Why did you make it fixed?

Comment (appcypher) [reply src/apis/emscripten/mod.rs]: I should probably remove it; this task doesn't depend on it. I was tryings things from the emscripten side.

File: src/apis/emscripten/storage.rs [hunk 2]
```diff
@@ -4,22 +4,15 @@ pub fn align_memory(ptr: u32) -> u32 {
...
+//     let old_static_top = *static_top;
+//     let total_memory = memory.maximum_size() * LinearMemory::PAGE_SIZE;
+//     // NOTE: The `4294967280` is a u32 conversion of -16 as gotten from emscripten.
+//     *static_top = (*static_top + size + 15) & 4294967280;
+//     assert!(
+//         *static_top < total_memory,
+//         "not enough memory for static allocation - increase total_memory!"
+//     );
+//     old_static_top
+// }
```
Comment (syrusakbary) [inline src/apis/emscripten/storage.rs]: Why is this necessary? Since I haven't created this, it might be good to get @lachlansneff feedback here

File: src/apis/emscripten/time.rs [hunk 3]
```diff
@@ -64,37 +66,137 @@ pub extern "C" fn _clock_gettime(clk_id: c_int, tp: c_int, instance: &mut Instan
...
+
+        let day = if date.tm_mday < 10 {"  "} else {" "};
+        let hour = if date.tm_hour < 10 {" 0"} else {" "};
+        let min = if date.tm_min < 10 {":0"} else {":"};
+        let sec = if date.tm_sec < 10 {":0"} else {":"};
+        let year = 1900 + date.tm_year;
+
+        let time_str = format!(
+            // NOTE: The 14 accompanying chars are needed for some reason
+            "{} {}{}{}{}{}{}{}{}{} {}\0\0\0\0\0\0\0\0\0\0\0\0\0\0",
```
Comment (syrusakbary) [inline src/apis/emscripten/time.rs]: Why so many `0`s?

Comment (appcypher) [reply src/apis/emscripten/time.rs]: Exactly! I'm still researching why it segfaults if they are not there. 
I'm going to share my concerns on the dev channel.

**Comment by syrusakbary** [comment]
@appcypher I've added a `emtests` folder with a specific test case for `localtime` (so it's easier to test)
